libs/GameSystems/DecreaseLifeTimeSystem.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+** EPITECH PROJECT, 2022
+** R-Type
+** File description:
+** DecreaseLifeTimeSystem
+*/
+
+#ifndef DECREASELIFETIMESYSTEM_HPP_
+#define DECREASELIFETIMESYSTEM_HPP_
+
+#include "World/World.hpp"
+#include "GameComponents/LifeTimeComponent.hpp"
+#include "GameComponents/DeathComponent.hpp"
+
+namespace ecs
+{
+    /// @brief This is the DecreaseLifeTime System. This System decrease LifeTime Component at every tick.
+    /// It's a system that will be called by the engine world with the function runSystem()
+    struct DecreaseLifeTime : public System {
+        inline void run(World &world) override final
+        {
+            std::vector<std::shared_ptr<ecs::Entity>> joined = world.joinEntities<Lifetime>();
+
+            auto move = [](std::shared_ptr<ecs::Entity> entityPtr) {
+                entityPtr.get()->getComponent<Lifetime>() = entityPtr.get()->getComponent<Lifetime>().timeLeft - std::chrono::duration<double>(1);
```

Comment:
Shouldn't we reduce by the elapsedTime shared resource value 
If it does -> create an issue to do it later (before the first delivery of course)

Reply:
We can reduce it by elapsed time but i don't have this shared resource